### PR TITLE
Allow configuring logger + its fields in library usage

### PR DESCRIPTION
**Pull request by mjungsbluth**
When using OPA as a library in a Golang project that also uses logrus for logging, the log messages generated from OPA might be missing context or are hard to understand for users. 

This PR adds the ability to configure a plugin manager's logger using an option and also changes the `WithFields`method of `sdk.logger` to inherit the fields of the base logger. This allows to pre-define common fields on the log output that get propagated to log messages further down.

### CODE REVIEW

**Review by anderseknert — APPROVED** [review]
Some overlap with the SDK PR, but this looks good to me!

File: plugins/plugins.go [hunk 1]
```diff
@@ -279,10 +286,15 @@ func New(raw []byte, id string, store storage.Store, opts ...func(*Manager)) (*M
...
 
+	if m.logger == nil {
+		m.logger = sdk.NewStandardLogger()
+	}
+
 	serviceOpts := cfg.ServiceOptions{
 		Raw:        parsedConfig.Services,
 		AuthPlugin: m.AuthPlugin,
 		Keys:       keys,
+		Logger:     m.logger,
```
Comment (tsandall) [inline plugins/plugins.go:297]: @anderseknert is there a reason we weren't passing this thru before? Was this just an oversight in the original PR?

**Review by tsandall — COMMENTED** [review]
Looks good to me. Just one question for @anderseknert about the change to set the logger on the cfg.ServiceOptions struct.

File: plugins/plugins.go (hunk 1 above)
Comment (anderseknert) [reply plugins/plugins.go:297]: Yeah looks like it was. The logger still gets attached to the config object when instantiating the `Client` [here](https://github.com/open-policy-agent/opa/blob/master/plugins/rest/rest.go#L168), but I believe that was part of an earlier iteration that should have been replaced by injecting it through the `ServiceOptions` rather, which looks like it only got half-done :/ Good to get that fixed! We shouldn't need to keep that previous way of doing it though, and that should be good to remove with this change.

**Pull request merged**